coderd/rbac/authz_test.go

```diff
@@ -229,7 +229,7 @@ func BenchmarkRBACAuthorizeGroups(b *testing.B) {
 
 // BenchmarkRBACFilter benchmarks the rbac.Filter method.
 //
-//	go test -bench BenchmarkRBACFilter -benchmem -memprofile memprofile.out -cpuprofile profile.out
+//	go test -bench '^BenchmarkRBACFilter$' -benchmem -memprofile memprofile.out -cpuprofile profile.out
```

Comment:
I think it makes sense to separate the `PrepareOnly` benchmarks into a dedicated test (e.g., `BenchmarkRBACPrepare`). The `Filter` method already calls `Prepare`, so it will always have a longer execution time than just `Prepare`. Mixing both leads to benchstat averages that can be misleading. See for example the benchstat results from the policy update PR: https://github.com/coder/coder/blob/bd86a8848c534637011a4d4d418818d6044ddd56/coderd/rbac/benchmarks/benchstat_filter_benchmark.txt#L25
Wdyt?

Reply:
That sounds reasonable to me!

Reply:
That is fine with me :+1: 

Ideally the `Filter` benchmark would omit the time used during `Prepare`? Then the total time is the combination of the benchmarks.

Regardless, feel free to make that refactor :+1: